docs/types/operators.rst

```diff
@@ -1,7 +1,34 @@
-.. index:: assignment, ! delete, lvalue
+.. index:: operator
 
-Operators Involving LValues
-===========================
+Operators
+=========
+
+Binary operators can be applied even if the two operands do not have the same type.
+For example, you can compute ``y = x + z``, where ``x`` is a ``uint8`` and ``z`` has
+the type ``int32``. In these cases, the following mechanism will be used to determine
+the type in which the operation is computed (this is important in case of overflow)
+and the type of the operator result:
+
+1. If the type of the right operand can be implicitly converted to the type of the left
+   operand, use the type of the left operand,
+2. if the type of the left operand can be implicitly converted to the type of the right
+   operand, use the type of the right operand,
+3. otherwise, the operation is not allowed.
```

Comment:
We could say that these are the generic rules, but they may be different for some types - check the documentation for the types to be sure.

Reply:
We could though it would be nice to have rules that are actually applies consistently. Is there a good reason for comparison between int and fixed-point not to work?

BTW, I don't see this quirk documented for the fixed-point type. It only says it provides the `==` operator.

Reply:
There is no such thing as `int_const`. `1.0` and `1` are absolutely identical.

Reply:
Oh, true.

Anyway, I still think this inconsistency should be documented somewhere.